filters/ColorinterpFilter.cpp

```diff
@@ -269,16 +285,33 @@ bool ColorinterpFilter::processOne(PointRef& point)
 {
     double v = point.getFieldAs<double>(m_interpDim);
 
-    // Don't color points that aren't in the min/max range.
-    if (v < m_min || v >= m_max)
+    if (m_clamp)
+    {
+         v = Utils::clamp(v, m_min, m_max);
+    }
+
+    // Don't color points that aren't in the min/max range
+    // unless they've been clamped. Allow v == m_max so that
+    // if the user wants to clamp all values outside m_min
+    // and m_max the values greater than m_max are colored
+    // as expected.
+    if (v < m_min || v > m_max)
+    {
         return true;
+    }
 
     double factor = (v - m_min) / (m_max - m_min);
     size_t img_width = m_redBand.size();
     size_t position = size_t(std::floor(factor * img_width));
 
+    // Handle the case that v == m_max (position == img_width) by clamping
+    // position to img_width - 1
+    position = std::min(position, img_width - 1);
```

Comment:
This needs to be guarded:

`(std::min)(position, img_width - 1);`

Sorry I didn't mention this before.